### PR TITLE
Build a portfolio web page.

### PR DESCRIPTION
**Pull request by hd-yang**
Build a portfolio web page.
Finish the task of week 1.

### CODE REVIEW

**Comment by hd-yang** [comment]
Check out the web page.
http://hyang-sps-summer20.appspot.com/index.html

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,93 @@
...
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  // private ArrayList<String> data = new ArrayList<String>();
```
Comment (akovski) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: clear debug code

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -19,14 +19,93 @@
...
+  //   response.getWriter().println("Hello Huidong Yang!");
+  // }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    ArrayList<String> data = loadData();
+    
+    // convert data to json
```
Comment (akovski) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Comment should start with capital letter to keep consistent style with following code

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -19,14 +19,93 @@
...
+    Gson gson = new Gson();
+    String json = gson.toJson(data);
+
+    // generate response
```
Comment (akovski) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: same here

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -26,3 +26,61 @@ function addRandomGreeting() {
...
+    console.log(data);
+    const datacontainer = document.getElementById('comments');
+    datacontainer.innerHTML = '';
+    for (i in data) {
+      datacontainer.appendChild(createListElement(data[i]));
+    }
+  });
+}
+
+/** Creates an <li> element containing text. */
```
Comment (akovski) [inline portfolio/src/main/webapp/script.js]: comments here also have inconsistent style, usually it should start with capital singular form

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -26,3 +26,61 @@ function addRandomGreeting() {
...
+    gameContainer.innerText = "Do you like <" + game[0] + "> ?";
+    gameImg.innerHTML = img_url;
+}
+
+// get data using fetch()
+function getData() {
+  fetch('/data').then(response => response.json()).then((data) => {
+    console.log(data);
+    const datacontainer = document.getElementById('comments');
+    datacontainer.innerHTML = '';
```
Comment (akovski) [inline portfolio/src/main/webapp/script.js]: please keep variable naming style consistent, either image_url or gameImg style, datacontainer may not be recommended

**Comment by hd-yang** [comment]
@akovski I have updated the code to fix these problems.

**Review by akovski — APPROVED** [review]

**Pull request merged**